### ISSUE TITLE
Error in label since 2.14.9

### ISSUE THREAD

**Issue by TheRoby** [pluginsGLPI/genericobject#395]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### GLPI Version

10.0.16

### Plugin version

2.14.10

### Bug description

Hello,
Since 2.14.9, the label of the new objet is the internal identifier ... (the <itemtype>.fr_FR.php was updated)

2.14.8 :
![image](https://github.com/user-attachments/assets/d9108c90-f97b-4fe3-944b-05b15fffe304)

2.14.9 or 2.14.10 : 
![image](https://github.com/user-attachments/assets/62c3188a-8ec1-45b5-a423-003da2fead9c)

same error if I create a new object.

Thanks

### Steps To reproduce

1. create a new object
2. change the libelle on the php file
3. refresh glpi ...

**Comment by orugari** [comment]
Still same problem, impossible to edit Label

**Comment by mykola-voroniak** [comment]
I encountered the same issue. I resolved it by adding "global $LANG;" to the .en_GB.php file.

![Image](https://github.com/user-attachments/assets/1a90f4a0-a014-45fd-a75a-31a04d35980f)

**Comment by orugari** [comment]
yes, this is not really clear on how to do it, but yes, thanks :)
It would be great to have an explanation, more useful on this array, for fields and labels. I tried somes, followed the guide, but no success for form labels.

**Comment by mykola-voroniak** [comment]
To change the label name, you need to go to glpi/files/_plugins/genericobject/locales/your_object/your_object.your_language.php.

In my case, it's glpi/files/_plugins/genericobject/locales/tool/tool.en_GB.php.

In this file, add global $LANG; and change the label name. Then, the object will be displayed as desired in the GLPI interface

**Comment by stonebuzz** [comment]
This may be related to a different handling of global variables in the latest versions of PHP.

I remember fixing the generation of this file from a template, as seen in the following commit:
👉 https://github.com/pluginsGLPI/genericobject/commit/d29683fc467938f17fd5dd08a65c1c7e0392b65a

The fix is still valid, but only relevant for objects created recently — roughly within the past 10 months since that PR.

As the translation file can be manually edited by developers, we chose not to modify files that were generated before this fix.
